### PR TITLE
Configure dev_appserver to avoid restarting modules on detected change

### LINKED ISSUE #836: Class changes during debugging should avoid reloading modules
With #609 we properly support WTP's auto-relaunching of changed modules.  This is good for changes to web resources, like HTML and .js files (and possibly .jsps?), but isn't great when making changes to Java files whose updated class files can be downloaded and hotswapped into the running JVM(s).

Complications include m2e-wtp which sets odd paths outside of the normal WAR structure for `.class` files, and necessitates our use of the publishing step.

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #836

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -8,7 +8,7 @@ Bundle-Localization: plugin
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
 Bundle-Activator: com.google.cloud.tools.eclipse.appengine.localserver.Activator
 Bundle-ActivationPolicy: lazy
-Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.1.0",
+Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.1.14",
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF]: Isn't it 0.1.13?

https://github.com/GoogleCloudPlatform/appengine-plugins-core/commits/v0.1.13

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java [hunk 2]
```diff
@@ -174,6 +174,7 @@ void startDebugDevServer(List<File> runnables, MessageConsoleStream console,
 
     // Create run configuration
     DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
+    devServerRunConfiguration.setAutomaticRestart(false);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java]: Why only for debug runs? I see that normal runs also restart on file changes.

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java]: (blush) um yes, good point.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF (hunk 1 above)
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF]: Mea culpa: I'm running against source.  [0.1.13 is in the wild](http://search.maven.org/#artifactdetails%7Ccom.google.cloud.tools%7Cappengine-plugins-core%7C0.1.13%7Cjar)

**Comment by briandealwis** [comment]
PTAL @chanseokoh

**Pull request merged**